test/common/quic/integration/BUILD

```diff
@@ -2,39 +2,44 @@ load(
     "//bazel:envoy_build_system.bzl",
     "envoy_cc_test",
     "envoy_package",
+    "envoy_select_enable_http3",
 )
 
 licenses(["notice"])  # Apache 2
 
 envoy_package()
 
+# TODO(mattklein123): Use target_compatible_with when we switch to Bazel 4.0 instead of multiple
+# selects.
 envoy_cc_test(
     name = "quic_protocol_integration_test",
     size = "large",
-    srcs = [
+    srcs = envoy_select_enable_http3([
         "quic_protocol_integration_test.cc",
-    ],
+    ]),
     data = ["//test/config/integration/certs"],
     shard_count = 8,
     tags = [
         "fails_on_clang_cl",
         "fails_on_windows",
         "nofips",
     ],
-    deps = [
-        "//source/extensions/quic_listeners/quiche:active_quic_listener_lib",
-        "//source/extensions/quic_listeners/quiche:client_connection_factory_lib",
-        "//source/extensions/quic_listeners/quiche:quic_factory_lib",
-        "//source/extensions/quic_listeners/quiche:quic_transport_socket_factory_lib",
+    deps = envoy_select_enable_http3([
```

Comment:
As commented above, some of the quic libraries are safe to include and others aren't.
Fine with this for now but do you think that in the long run we should make it safe to include all the libraries, and bazel select the libraries to be no-ops for non-quiche?